vnext/Chakra/ChakraJsiRuntime.h

```diff
@@ -0,0 +1,241 @@
+#pragma once
+
+#if defined(USE_EDGEMODE_JSRT)
+#include <jsrt.h>
+#else
+#include <ChakraCore.h>
+#endif
+
+#include "ChakraJsiRuntimeArgs.h"
+
+#include <jsi/jsi.h>
+
+#include <memory>
+#include <mutex> 
+#include <sstream>
+#include "UnicodeConversion.h"
+
+namespace facebook { 
+namespace jsi { 
+namespace chakraruntime {
+
+class ByteArrayBuffer : public jsi::Buffer {
+public:
+  size_t size() const override {
+    return size_;
+  }
+
+  const uint8_t* data() const {
+    return byteArray_.get();
+  }
+
+  uint8_t* data() {
+    return byteArray_.get();
+  }
+
+  ByteArrayBuffer(int size) : size_(size), byteArray_(std::make_unique<uint8_t[]>(size)) {}
+
+private:
+  int size_;
+  std::unique_ptr<uint8_t[]> byteArray_;
+};
+
+class ChakraJsiRuntime : public jsi::Runtime {
+public:
+  ChakraJsiRuntime(ChakraJsiRuntimeArgs&& args) noexcept;
+  ~ChakraJsiRuntime() noexcept;
+
+  void evaluateJavaScript(std::unique_ptr<const jsi::Buffer> buffer, const std::string& sourceURL) override;
+  jsi::Object global() override;
+
+  std::string description() override;
+
+  bool isInspectable() override;
+
+  void setDescription(const std::string& desc);
+
+  // JsValueRef->JSValue (needs make.*Value so it must be member function)
+  jsi::Value createValue(JsValueRef value) const;
+
+  // Value->JsValueRef (similar to above)
+  JsValueRef valueRef(const jsi::Value& value);
+
+protected:
+  ChakraJsiRuntimeArgs& runtimeArgs() {
+    return args_;
+  }
+
+private:
+  static void CALLBACK PromiseContinuationCallback(JsValueRef funcRef, void* callbackState) noexcept;
+  void promiseContinuation(JsValueRef value)  noexcept;
+
+  void setupNativePromiseContinuation() noexcept;
+  void setupMemoryTracker() noexcept;
+
+private:
+  // Arguments shared by the specializations ..
+  ChakraJsiRuntimeArgs args_;
+
+  // Note :: For simplicity, We are pinning the script and serialized script buffers in the JSI::Runtime instance assuming as these buffers 
+  // are needed to stay alive for the lifetime of the JSI::Runtime implementation. This approach doesn't make sense
+  // for other external buffers which may get created during the execution as that will stop the backing buffer from getting released
+  // when the JSValue gets collected.
+
+  // These buffers back the external array buffers that we handover to ChakraCore.
+  std::vector<std::shared_ptr<const jsi::Buffer>> pinnedPreparedScripts_;
+
+  // These buffers are kept to serve the source callbacks when evaluating serialized scripts.
+  std::vector<std::shared_ptr<const jsi::Buffer>> pinnedScripts_;
+
+private:
+  static std::once_flag s_runtime_version_init_flag_;

```

Comment:
>s_runtime_version_init_flag_ [](start = 24, length = 28)

nit s_runtimeVersionInitFlag to conform to the convention of other names.

Reply:
Unified many names to MS/Office conventions ..